feat(docs): update how we run checks with the new retries flow [sc-14873] (#789)
checkly/checklyhq.com: site/content/docs/monitoring/_index.md

There are two reasons for this:

1. Redundancy: we might have an issue in one location, but not the other. 

> 2. Double checking: if your check fails and you have "double-check" enabled we will retry the check from the other location.

Reviewer: ```suggestion
2. Double checking: if your check fails and you have "double-check" enabled we will retry the check from another location.
```

1. A cron process picks up a check based on its schedule, say every 5 minutes. It validates that the check is not in progress at the moment to avoid race conditions. The check is put into a queue to be run from the next configured data center location.
2. If the check is an API check and has a [setup script](/docs/api-checks/setup-teardown-scripts/), the setup script is executed. 
3. The check is executed.
4. If the check is an API check and has a [teardown script](/docs/api-checks/setup-teardown-scripts/), the teardown script is executed.
Teardown scripts are run *before* any assertions are validated.
5. The result is stored in our central database.
6. If the check fails and “double-check” is enabled, the process starts over from a different location. The other location is picked, at random, from all the configured locations. If only one location has been selected, then the other location is picked at random from all available locations. Any setup & teardown scripts are run again as part of the process.

> 7. Alerts are sent out in requested channels when the sequence is complete. It's considered complete when the check run was successful or the final attempt was executed. We will send alerts only if the final attempt has failed (no alerts sent for the initial attempts)

Reviewer 2: ```suggestion
7. When the sequence is complete, we send alerts to the channels as requested. A sequence is considered complete if either the check run was successful or the final attempt was executed. Please note that we will only send alerts if the final attempt has failed. No alerts will be sent for the initial attempts.
```

Asked ChatGPT to improve the text.

1. A cron process picks up a check based on its schedule, say every 5 minutes. It validates that the check is not in progress at the moment to avoid race conditions. The check is put into a queue to be run from the next configured data center location.
2. If the check is an API check and has a [setup script](/docs/api-checks/setup-teardown-scripts/), the setup script is executed. 
3. The check is executed.
4. If the check is an API check and has a [teardown script](/docs/api-checks/setup-teardown-scripts/), the teardown script is executed.
Teardown scripts are run *before* any assertions are validated.
5. The result is stored in our central database.
6. If the check fails and “double-check” is enabled, the process starts over from a different location. The other location is picked, at random, from all the configured locations. If only one location has been selected, then the other location is picked at random from all available locations. Any setup & teardown scripts are run again as part of the process.

> 7. When applicable, alerts are send out in requested channels.

Reviewer 2: I think we need to explain here when the whole sequence finishes.
* check passed, or the final attempt was executed
* only then we do send alerts (not for initial attempts)

Author: @jan-osch Done! Lemme know if it's good.
